src/main/kotlin/com/cognifide/gradle/aem/instance/InstanceSync.kt

```diff
@@ -297,6 +300,59 @@ class InstanceSync(val project: Project, val instance: Instance) {
         return response
     }
 
+    fun downloadPackage(remotePath: String, targetDir: File): File {
+        lateinit var exception: FileException
+        for (i in 0..config.downloadRetryTimes) {
+            try {
+                return downloadPackageOnce(remotePath, targetDir)
+            } catch (e: FileException) {
+                exception = e
+
+                if (i < config.downloadRetryTimes) {
+                    logger.warn("Cannot download package $remotePath from $instance.")
+                    logger.debug("Download error", e)
+
+                    val header = "Retrying download (${i + 1}/${config.downloadRetryTimes}) after delay."
+                    val countdown = ProgressCountdown(project, header, config.downloadRetryDelay)
+                    countdown.run()
+                }
+            }
+        }
+
+        throw exception
+    }
+
+    fun downloadPackageOnce(remotePath: String, targetDir: File): File {
```

Comment:
There can be a number of things that can go wrong. Especially with bigger content packages on remote locations. (broken connection, 404 on initial request because AEM didn't list the package immediately after build etc.)  I would rather keep it here.

Reply:
ok, fine

Reply:
still align retry to new approach / use `AemConfig.Retry`